DigiMeSDK/Core/Classes/DMEPullClient.m

```diff
@@ -273,21 +274,24 @@ - (void)evaluateSessionDataFetchProgress:(BOOL)schedulePoll
         NSLog(@"DigiMeSDK: Sync status - %@", self.sessionFileList.syncStatusString);
     }
     
-    if (![self syncRunning] && !self.apiClient.isDownloadingFiles)
+    // If sessionError is not nil, then syncStatus is irrelevant, as it will be the previous successful fileList call.
+    if ((self.sessionError != nil || ![self syncRunning]) && !self.apiClient.isDownloadingFiles)
     {
```

Comment:
Before this fix SDK will stop polling server on any error within 3 sec. Within this fix you added another 3 sec delay before SDK will stop retrieving files by introducing `NSError *sessionError` to store an error and then SDK will cancel poll within next attempt, so 3+3sec. I still think that it is not enough for this specific place, and the same issue will appear in the future in my opinion.

It is very easy to reproduce this issue because everything is happening on a mobile device where network problems is a normal situation. However, It is absolutely correct behaviour to abort the flow and return an error immediately when USER is **interacting** with the app. Because nobody wants to wait while working. But in this particular place user has done all interactions and now SDK is trying to deliver all requested JFS files. This is polling part like a **delivery service** or a pipe... When you sending a photo using iMessage or WhatsApp or when you are using Dropbox to sync your files - all this services are more reluctant to do the abort immediately. They are trying and trying, and only after some delay they will abort. Even when SDK is trying to download any JFS file it has 5 attempts by default...